src/backend.rs

```diff
@@ -231,6 +231,23 @@ pub enum KeyState {
     Pressed,
 }
 
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub struct TouchPosition {
+    pub x: Fixed,
+    pub y: Fixed,
+    pub x_transformed: Fixed,
+    pub y_transformed: Fixed,
+}
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub enum TouchEvent {
+    Down { pos: TouchPosition },
+    Up,
+    Motion { pos: TouchPosition },
+    Cancel,
+    Frame,
+}
```

Comment:
This should be flattened into the InputEvent enum and the fields of TouchPosition should be flattened into the Down and Motion variants.